astropy/convolution/convolve.py

```diff
@@ -163,15 +163,8 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
         array_internal = np.array(array, dtype=float)
         array_dtype = array_internal.dtype
     elif isinstance(array, np.ndarray):
-        # Note this won't copy if it doesn't have to -- which is okay
```

Comment:
You could leave the first sentence in because it's kind of important that nothing should change `array_internal`.